x-pack/plugins/uptime/public/components/monitor/ping_list/columns/ping_timestamp/step_image_caption.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiText } from '@elastic/eui';
+import React from 'react';
+import moment from 'moment';
+import { nextAriaLabel, prevAriaLabel } from './translations';
+import { getShortTimeStamp } from '../../../../overview/monitor_list/columns/monitor_status_column';
+
+export interface StepImageCaptionProps {
+  captionContent: string;
+  imgSrc?: string;
+  maxSteps?: number;
+  setStepNumber: React.Dispatch<React.SetStateAction<number>>;
+  stepNumber: number;
+  timestamp: string;
+}
+
+export const StepImageCaption: React.FC<StepImageCaptionProps> = ({
+  captionContent,
+  imgSrc,
+  maxSteps,
+  setStepNumber,
+  stepNumber,
+  timestamp,
+}) => {
+  return (
+    <>
+      <div className="stepArrowsFullScreen">
+        {imgSrc && (
```

Comment:
I was trying to avoid making any direct modifications to rendering logic as this was largely intended as a refactor, outside of the additional change to resolve https://github.com/elastic/kibana/issues/80173.

This is related to your above comment about `maxSteps` as well. All of this weirdness happens because `useFetcher` can return `null | undefined` for data. Perhaps we should require all props for these child components, and let a top-level `data is truthy` assertion dictate whether we even try to render any of them.

`maxSteps`, for example, appears to have a default value. So unless `data` itself is not returned, we should have values for these fields.